.github/workflows/ci-linux.yaml

```diff
@@ -4,36 +4,67 @@ on: [push, pull_request]
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
-        os: [ubuntu-latest]
 
     steps:
       - name: Checkout source
         uses: actions/checkout@v3
         with:
           submodules: recursive
 
-      - name: Set up Python
-        uses: actions/setup-python@v4
+      - name: Set up Conda
+        uses: conda-incubator/setup-miniconda@v2.2.0
         with:
-          python-version: ${{ matrix.python-version }}
+            channels: conda-forge
+            python-version: ${{ matrix.python-version }}
+
+      - name: Show info about `base` environment
+        shell: "bash -l {0}"
+        run: |
+          conda info
+          conda config --show-sources
+          conda list --show-channel-urls
+
+      - name: Set up `env`
+        shell: "bash -l {0}"
+        run: >
+          conda create -n env
+          c-compiler cxx-compiler
+          python=${{matrix.python-version}} wheel pip
+
+      - name: Show info about `env` environment
+        shell: "bash -l {0}"
+        run: |
+          conda list --show-channel-urls -n env
 
       - name: Install numcodecs
+        shell: "bash -l {0}"
         run: |
+          conda activate env
+          export DISABLE_NUMCODECS_AVX2=""
```

Comment:
This was added since there were issues compiling with AVX2 on Linux. Would be good to revisit why this is occurring. That said, this is already [how we build in conda-forge]( https://github.com/conda-forge/numcodecs-feedstock/blob/ad779a7eaa6efcfe2ea54f8f0ba637d87ebaf1a3/recipe/meta.yaml#LL16 ).